ethpm/contract.py

```diff
@@ -105,6 +93,20 @@ def validate_attr_dict(self, attr_dict: Dict[str, str]) -> None:
                 )
 
 
+def is_prelinked_bytecode(bytecode: bytes, link_refs: List[Dict[str, Any]]) -> bool:
+    """
+    Returns False if all expected link_refs are unlinked, otherwise returns True.
+    todo support partially pre-linked bytecode (currently all or nothing)
+    """
+    for link_ref in link_refs:
+        for offset in link_ref["offsets"]:
+            try:
+                validate_empty_bytes(offset, link_ref["length"], bytecode)
```

Comment:
This is *probably* an **ok** temporary solution, but I don't think that checking for all `0`'s in the link locations should be the final solution since it is valid to link against the zero address.  The *final* solution should probably be cross checking between `link_dependencies` and `link_references` and seeing which elements from the `link_references` are not satisfied in the `link_dependencies`  (it's possible I've reversed those two, I get confused as to which one represents the "it was linked with these values" and which one is "these are the places that need linking".)

Reply:
Agreed, put it on the list - will work on improving this in a future pr